semgrep/semgrep/core_output.py

```diff
@@ -196,53 +198,61 @@ def parse(cls, raw_json: JsonObject) -> "CoreSkipped":
 
 
 @attr.s(auto_attribs=True, frozen=True)
-class CoreTargetTiming:
-    rule_id: RuleId
-    target: Path
+class CoreRuleTiming:  # For a given target
+    rule: Rule
     parse_time: float
     match_time: float
-    run_time: float
 
     @classmethod
-    def parse(cls, raw_json: JsonObject, rule_id: RuleId) -> "CoreTargetTiming":
-        # rule_id = RuleId(raw_json["rule_id"])
-        path = Path(raw_json["path"])
+    def parse(
+        cls, rule_table: Dict[RuleId, Rule], raw_json: JsonObject
+    ) -> "CoreRuleTiming":
+        rule = rule_table[RuleId(raw_json["rule_id"])]
         parse_time = raw_json["parse_time"]
         match_time = raw_json["match_time"]
-        run_time = raw_json["run_time"]
-        return cls(rule_id, path, parse_time, match_time, run_time)
+        return cls(rule, parse_time, match_time)
 
 
 @attr.s(auto_attribs=True, frozen=True)
-class CoreRuleParseTiming:
-    rule_id: RuleId
-    parse_time: float
+class CoreTargetTiming:
+    target: Path
+    per_rule_timings: List[CoreRuleTiming]
+    run_time: float
 
     @classmethod
-    def parse(cls, raw_json: JsonObject, rule_id: RuleId) -> "CoreRuleParseTiming":
-        # rule_id = RuleId(raw_json["rule_id"])
-        parse_time = float(raw_json["rule_parse_time"])
-        return cls(rule_id, parse_time)
+    def parse(
+        cls, rule_table: Dict[RuleId, Rule], raw_json: JsonObject
+    ) -> "CoreTargetTiming":
+        target = Path(raw_json["path"])
+        per_rule_timings = [
+            CoreRuleTiming.parse(rule_table, timing)
+            for timing in raw_json["rule_times"]
+        ]
+        run_time = raw_json["run_time"]
+        return cls(target, per_rule_timings, run_time)
 
 
 @attr.s(auto_attribs=True, frozen=True)
 class CoreTiming:
```

Comment:
Similarly to comment above, IMO CoreTiming should just be timing json returned by semgrep-core and it is up to ProfileManager class to combine list of Rules and CoreTiming

Reply:
Leaving as is for now, this is a more minimal change from the original